source/patterns/@aws-solutions-constructs/core/lib/opensearch-defaults.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ *  Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import * as opensearch from 'aws-cdk-lib/aws-opensearchservice';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as cdk from 'aws-cdk-lib';
+import { BuildOpenSearchProps } from './opensearch-helper';
+
+export function DefaultOpenSearchCfnDomainProps(domainName: string, cognitoConfigureRole: iam.Role, props: BuildOpenSearchProps):
+  opensearch.CfnDomainProps {
+  const roleARNs: iam.IPrincipal[] = [];
+
+  roleARNs.push(new iam.ArnPrincipal(props.cognitoAuthorizedRoleARN));
+
+  if (props.serviceRoleARN) {
+    roleARNs.push(new iam.ArnPrincipal(props.serviceRoleARN));
+  }
+
+  return {
```

Comment:
Can we cite a doc link in a comment here that supports these defaults? (I know that's new, but it would be nice)